src/fairchem/core/trainers/base_trainer.py

```diff
@@ -277,7 +277,7 @@ def load_datasets(self) -> None:
         self.test_loader = None
 
         # load train, val, test datasets
-        if self.config["dataset"].get("src", None):
+        if self.config.get("dataset", None) and self.config["dataset"].get("src", None):
```

Comment:
simpler? if "src" in self.config.get("dataset")?

Reply:
Same as below. Some of our configs have config['dataset']=None, so that the key is present but it is None.